clicommand/tool_keygen.go

```diff
@@ -0,0 +1,134 @@
+package clicommand
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+
+	"github.com/buildkite/agent/v3/internal/jwkutil"
+	"github.com/lestrrat-go/jwx/v2/jwa"
+	"github.com/urfave/cli"
+	"golang.org/x/exp/slices"
+)
+
+type KeygenConfig struct {
+	Alg                   string `cli:"alg" validate:"required"`
+	KeyID                 string `cli:"key-id" validate:"required"`
+	PrivateKeySetFilename string `cli:"private-keyset-filename" normalize:"filepath"`
+	PublicKeysetFilename  string `cli:"public-keyset-filename" normalize:"filepath"`
+
+	NoColor     bool     `cli:"no-color"`
+	Debug       bool     `cli:"debug"`
+	LogLevel    string   `cli:"log-level"`
+	Experiments []string `cli:"experiment"`
+	Profile     string   `cli:"profile"`
+}
+
+var KeygenCommand = cli.Command{
+	Name:  "keygen",
+	Usage: "Generate a new JWS key pair, used for signing and verifying jobs in Buildkite",
+	Description: `Usage:
+
+    buildkite-agent tool keygen [options...]
+
+Description:
+
+This (experimental!) command generates a new JWS key pair, used for signing and verifying jobs in Buildkite.
+The key pair is written to two files, a private keyset and a public keyset. The private keyset should be used
+as for signing, and the public for verification. The keysets are written in JWKS format.
+
+For information about signing jobs in buildkite, see: <PUT A LINK HERE YA DINGO>
```

Comment:
@triarius how do you feel about a <PUT A LINK HERE YA DINGO> going into main? we don't have docs sorted for signed pipelines just yet...

Reply:
maybe i'll update it to say "`<PUT A DOC LINK HERE WHEN THERE IS ONE>`"

Reply:
maybe i'll update it to say "`<PUT A DOC LINK HERE WHEN THERE IS ONE>`"

Reply:
Delete the line an put a TODO comment above the varaible?

Reply:
much more sensible.